### PR TITLE
v1.3 compatibility

### PR DESCRIPTION
**Pull request by njzjz**
- implement projection in one unit. other descriptors does not work due to the change of interface
- add missing file
- backup, seperate para and vert descriptors
- make the testing code work. option to shift the total energy zero
- add rotation test
- fix bug
- fix bug of initializing base class DescrptSeA
- first working version of type embedding
- simplify the implementation
- implement one-side type embedding, fix bug of type normalization
- work around the gcc bug
- fix bug of pass_filter interface
- first working version of aparam embedding
- add tensorboard support! visualize the training of deepmd-kit
- fix potential bugs
- Add tensorboard tutorial
- implement external efield in fix/dplr
- refactorize the embedding net
- first implementation of 3-body embedding
- consistent indent, fix bug
- clean up the code
- fix bug. improved angle embedding
- implement hybrid descrpt
- add unittest for embedding net
- add doc for short-range tabulated interaction
- fix bug of missing model attribute in hybrid descriptor
- small changes to minimize PR diff
- added ommited tf summaries in some of the descriptors
- update the doc for descriptor type
- fix bugs in argcheck
- fix typos
- reworked TB logdir clean system
- remove forgotten experimental tweaks
- first working version
- fixed bug in dipole fitting
- fix bug of compulsory key `loss`
- fix bug of compulsory key `loss`
- fix bug of compulsory lable requirement
- fix bug of compulsory lable requirement
- fix bugs
- add the instruction to install tf 2.3
- fix most of the descriptors
- Update install.md
- add descrpt hybrid and loc_frame
- add python inference
- add loss, learning rate, tabulated interaction, data modifier
- add data and data_system
- install the package in a pythonic way
- fix bug: removed folder api
- move common, env, compat to deepmd and deepmd/utils
- avoid `using namespace std` in all c++ code
- rm unnecessary file
- change 0 -> 0.0 when providing a float variable
- fix bug of nbor sorting
- use lammps max_nbor_size as the upper boundary of gpu sorting
- fix a potential bug
- fix bug of nbor sorting
- fix conflicts of cherry-pick
- fix a potential bug
- fix bug of std::std::
- fix bugs of std error
- model compression
- Update deepmd/utils/tabulate.py
- migrate from travis-ci to github actions
- add package tqdm into setup.py
- add the reference of global_np_float_precision into common.py
- optimize the performance of gpu implementations of custome ops
- through compression error when resnet_dt is set true for descriptor
- update cmake support for Anpere architecture devices
- add dosctings, pathlib, typing, uppercase constants to few base files
- updates to common
- rewrite config script
- small fixes
- frezer script rewrite
- enhancements to calculator
- debug git action cmake build
- install tf in git actions since it is not found
- fix failing tests
- update compiler support for Ampere architecture devices
- fix github actions
- fix setup.py
- fix github actions
- add python lint git action
- failing to resolve used action repo
- fix pyaction
- changes to test the lintner action
- bugfixes to lint action
- add missing requirements and python 3.8 tests
- optimize the code structure of model compression
- add requirements file
- trigerr new actions lint
- trigger actions lint
- change the class name from DataInfo to EnvMatStat
- optimize the interface of EnvMatStat class
- use standard tensorflow op style for custome ops
- mv the c++ API to api_cc. Cleaning up the code for formatting neighbor list and building environment matrix. Add unittest for the lib
- add requested relative paths
- optimize code structure for method EnvMatStat
- Update CustomeOperation.h
- test new liner action
- fix bug of missing lib file when installing python interface
- clean up env_mat_r, add unittests
- fix bug of unchanged env_mat_r
- debug lint
- debug actions lint
- optimize code structure of model compression
- optimize code structure of model compression
- fix bug in compiling. fix bug in fmt_nlist_cpu: wrong handeling of too small sel
- refact the prod_force
- move the table precision control into descriptor
- Update use-deepmd-kit.md
- add intros of model compression to the README
- refact prod virial. improved interface of prod_force_a
- refact prod_force_r and prod_virial_r. add unittests
- rework runoptions, cluster, rename modlues to camel_case and constants to UPPERCASE
- move cluster folder to deepmd package
- remove readme now it is in freeze.py references
- rework entry point
- add new logging logic based on python logging
- typo fix
- fix bugs
- small fixes
- refact prod_force_grad prod_virial_grad soft_min, add unittests
- add type hint and mute warning message
- add type hint for class NrighborStat
- refact prod_env_mat soft_min_switch_force. fix issues in soft_min_switch
- bugfix
- rm print in class NeighborStat
- refact soft_min_swich_virial, soft_min_swich_force_grad, soft_min_swich_virial_grad
- changed NeighborList to neighbor_list. changed some miss leading names in prod_env_mat
- send model file through mpi bcast
- fix a bug, change the default memory size of temporary tensor
- rename tab_inter -> pair_tab ; fix bug introduced by changing NeighborList -> neighbor_list
- print verbose debug info
- small bugfixes
- workflow fix after deps moved to requirements.txt file
- test file bugfix
- fix lintner action
- debug weird actions fail
- debug failing build
- debug failing tests
- fix cluster subpackage not included
- bugfix
- update lint action
- try to recover the old setting of building libop
- mv the mpi API to the lammps package
- propose coding guidelines
- refact pair_tab
- reuse function get_file_content
- fix bug of free memory
- avoid copied code in pair_tab
- fix bug of lammps make serial error
- change name pair_tab -> pair_tab_cpu
- refact map_aparam
- fix bug: missing include file
- fix bug typo of variable name
- refact prod_force_grad_se_r_grad and prod_virial_se_r_grad
- move entrypoint scripts to one submodule
- install ops to specific module
- fix bugs after ops move
- remove forgoten debug prints
- add setup cfg and scilence some lintner errors
- small style fixes
- add strings formating suggestions
- whitespace formating guidelines
- implements factory pattern for Deep potentials
- bugfix in for log hnadles
- add pathlib option to potential factory
- get rid of args object passed to entrypoint scripts, it is not transparent. Unpack arguments instead and send them as keywords. Rewrite entrypoint scripts to reflect that
- rewrite transform and train enrtrypoints
- remove eaccidentally commited files
- migrate to pytest
- remove failed tests and report coverage
- set cov path
- change cov dir
- use develop mode
- running tests in the root directory
- fix errors
- fix bug
- fix bug
- convert PosixPath to string
- +1
- adjust the interface of custome op
- better format and implementation
- add unittests for deepmd.infer
- fix bug of wrong test data path
- change prints in transform to log
- add unittest for transform
- remove duplicated code in test_transform
- fix op module not being found
- fix DeepPotential factory tests
- remove forgotten debug prints
- added log level setting for tensorflow and OpenMP
- add unittest for TestEnvMatA gpu implementation
- adjust indent format of descrpt_se_a_multi
- adjust code structure of custom op
- remove global path of polar examples.
- change the files name in 'lib/include'
- Add introduction about the acceptable original model version in model compression part
- update model compression document
- fix known bugs
- install tensorflow_cc automatically with cmake and conda
- fix syntax error
- test both two FLOAT_PRECs
- add C++ unittest to CI
- fix the script
- added tests for argument parser
- optimize interface of custom ops
- fix bug in expand_sys_dir, add unittest for it
- adjust code spaces
- fix bug in main (no command) and config (systems type and rcut default type)
- fix bug in neighbor stat
- rm custom_op.h from api_cc to op
- rm compute_t as well as HIPH_PREC from custom_op.h
- rm MathUtilities.h and mv all necessary functions to utilities.h
- rm api_cc/include directory from op/CMakeLists.txt
- fix bugs
- test building gpu version C++ API
- fix action bugs
- fix bugs of using dot3
- clone submodules
- fix bug in log levels parser
- print message to std:cerr and return rather than assertion. The assertions will be bypassed in release building mode
- change name of format_nlist to format_nlist_i, because the function only deals with one of the atoms
- clean up the cpu interface of env_mat and fmt_nlist.
- fix bug of max_nbor_size usage
- Update CMakeLists.txt
- c++ style interface for format_nlist. independent impl. of Region and UT.
- small fix
- changed ubuntu latest to 18.04 in gh actions
- implment coord and neighbor_list, add unittests
- change python test OS to ubuntu-18.04
- add missing header file
- small fixes to parser
- add unit test for custom ops
- fix conflicts
- changing neighbor list interface. add unittests for op prod_env_mat
- initialize cpu memory before it copies to the device
- CPU implementation of op prod_env_mat
- shutdown the verbose debug information in test
- cleanup the code. switch to prod_env_mat_a in training
- add old style nlist vars. Waiting for the update of gpu part
- better impl. call prod_env_mat_a only once
- adjust the UT of prod_env_mat_r to new input nlist
- fix bug of illegal device memory access
- fix bug of cmake script for finding tensorflow
- change pb files to test ProdEnvMat
- switched to new nlist interface. clean up c++ api. add unittests.
- add destructor for InputNlist
- freeze output_dim in tensor models
- DipoleChargeModifier determine number of atoms by input vector of atom types
- clean up outdated commented code
- add test CI for api_cc
- install tensorflow
- set INSTALL_PREFIX
- change the directory of version.h
- clean up model devi, add unittests
- add UT for model devi in the atomic case
- fall back to some old name convention
- cd api_cc/tests
- implement prod_env_mat_r on cpu, add unittests
- fix bug in prod_env_mat. python infer evaluate a batch rather than frame by frame
- add lammps nbor_list support for gpu inference
- fix conflicts
- fix conflicts
- update names of the c++ api
- Reduce the num of device memory copies
- fix bug of ewald recp: transposed recp box
- clean up include of SimulationRegion. Add UT for num diff of force and virial
- fix bug in test_ewald
- clean up ewald, remove dependency on SimulationRegion
- fix bug in lib UT
- add model version support. implemented python interface.
- support model version check in c++ interface
- improved cmake message for model version
- print model version on summary string
- fix bugs in UT
- fix bug of model path in UT
- include the c++ api with deepmd prefix
- add deepmd namespace for the c++ interface
- add deepmd namespace for lib
- fix bug in tests
- fix bugs in multi_device ops
- namespace for tabulate and utilities
- fix bug : namespace in cuda compiling
- fix bugs of single precision training and transfer
- fix bug of nbor sorting
- fix bug of gpu namespace
- add namespace deepmd for gpu_cuda.h
- better output format during training, testing and in lcurve.out
- fix bugs of ase calculator and add tests
- add ase to tests
- fix typo
- fix bug
- import Path for runtime
- refact model
- fix bug of getting inputs
- fix bug of UTs and srtab
- fix bug of gen tab.xvg
- prod_env_mat gpu update
- fix bugs of tensor model
- add module train. mv some global vars from run_opt to env
- do not install moved files
- mv main to entrypoints, mv source/train to source/config
- fix bugs
- fix bug
- fig bug: wrong place of model version
- rm old-style implementation of prod_env_mat, prod_force and prod_virial
- fix bug in cmake config of api UT
- fix bug in multi frame support
- fix bugs of calling old OPs
- fix bug in UT
- fix bug of using old se_r
- gpu region update
- format update
- add region.cuh
- Update install.md
- Update use-deepmd-kit.md
- Update index.rst
- Update index.rst
- Create novel-auxiliary-options
- Update index.rst
- Update use-deepmd-kit.md
- Adding entries in document
- doc modification strike 2
- doc modification strike 3
- doc modification strike 4
- install g++-7 in GitHub Actions
- changed the unit of virial
- update doc for intallation
- install g++-7 in GitHub Actions
- prod_force_grad&prod_virial_grad gpu update
- fix bug in prod_env_mat.cu
- fix bug in prod_env_mat.cu
- update dipole training and test: support combination of glocal and atomic dipole
- final check of updated dipole mode, few modification
- few modifaction, compatile with polar mode
- add options `sys_probs` and `auto_prob_style`, fix bug in sys_probs
- ut test debug: change input argument atomic-energy to atomic
- PR debug: take Han Wang's advice, not to change the value of weight; change the name of the weight to pref_dipole and pref_atomic_dipole
- training on all sets in the training data systems. strict argcheck.
- set as default training data excludes "test set".
- update the argcheck to use the auto generated ref for variants from dargs
- test removed; validation supported.
- fix bugs in examples. unsed key does not pass argcheck in strict mode
- add doc for modifier
- fix bugs: 1, exclude type in ebd. 2, remove se_ar, use hybrid instead
- fix bugs. remove outdated examples
- use new reference style in gen_doc
- clean up doc for coding conventions
- cheating the argcheck for debugging
- variable names optimized; get_batch function optimized.
- minor bug fixed; data cfg scheme altered.
- roll back to incorp. normalization.
- stress dargs.RAW_ANCHOR
- minor bug fixed
- minor bug fixed
- update requirement of dargs
- dargs version corrected.
- default prec. changed to high
- none validation supported; get_feed_dict optimized.
- get_evaluation_rst altered to derive atomic avg.
- atomic avg. derived in a compromised way.
- seperate impl of averging errors across systems
- change l2 error to rmse
- fix bug of returned key
- bug fixings: 1, when valid data is not given; 2, bug related to iter over a dict.  Not using ClassArg in trainer anymore
- add default values for trainer settings, in case we do not have argcheck normalization
- print data summary with notation on the data systems
- printing the training error with exactly the batch used to train the model
- fix 2 spacing issues
- more understandable variable names
- fix bug in se_t, add unittest
- fix bug in lcurve.out output
- clean compat v1 -> v2 implemented
- change name : se_a_t -> se_t
- fix bug in ut
- compat v1 -> v2 unittest implemented
- fix bug in ut: duplicated graph name
- test bug fixed
- change names of descriptors, old names are still compatible
- import fixed
- normalized the list items of descriptor `hybrid`
- update doc: changed name of descriptors. with updated dargs
- update the doc and examples
- update git ignore
- YWolfeee: modified to support loss normalization
- fix bugs in examples
- add what's new in v2.0 to readme. move troubleshooting to a separate page
- use bit operations to construct nbor sort keys
- fix bug of using old name se_a, changed to se_e2_a
- fix bugs in UT
- add ut for encoding/decoding nbor info
- close loss normalization, since it need further examination
- Fix bug in coord.cu
- Add lib coord unittest
- refact fit/ener.py, add UT for atom_ener, fix zero divider bug in se_a
- add support for type embedding in fitting. fix bugs in fit/ener.py
- extend the input dim of embedding net
- add support for type embedding
- update doc for input parameters
- add doc for type_embed, fix bug
- add UT for type_one_side
- fix bug: type embedding should be optional
- UT for std calculations
- refact the impl. of std and relative std
- impl model deviation of virial
- support relative model deviation for virial
- fix bug: declaration of all_virial should be available to all precisions
- fix issue #598
- test lammps builds
- copy lammps plugin
- Re-structured Doc (#513)
- add ut for model compression (#586)
- Merge Tensor Loss Training and Polar Normalization (#593)
- print the virial model deviations in the place of zeros (#606)
- fix bug of type embed
- fix bug of type embedding
- optimize UT for dp transfer
- fix bug in test_transfer.py
- add coverage report for c/c++ api
- uploading once is enough
- fix bug of issue/580
- Fix bug in tf allocate_temp
- change options
- Apply suggestions from code review
- link coverage_config
- Ignore tests
- fix TB logdir remove bug
- add UUID support for DP
- check validity of data systems. print help message
- add Important hint at getting-start.md (#622)
- add doc of type embedding (#625)
- Optimized mkindex function in doc/conf.py and added two files in troubleshooting. (#619)
- support MPI and other atom_styles for LAMMPS atomic keyword (#628)
- fix spell mistake (#638)
- Readme and Examples for Tensor mode (#632)
- append to out_file when LAMMPS restarts (#640)
- add an example of C++ inference to doc (#652)
- Add instructions for i-PI (#660)
- Added doc for netsize setting, num_nodes specification, and sel setting in doc/troubleshooting/ (#657)
- fix issue 668 (#680)
- fix bug in op definition prod_virial_grad_multi_device.cc.  (#685)
- Add ROCM support for DEEPMD (#656)
- Fix uniform random seed (#691)
- Doc api cc (#663)
- add op unittests for prod_force, prod_virial, prod_force_grad and prod_virial_grad (#703)
- fix bug of adding int to a None random seed (#705)
- update the doc for training input (#706)
- add v1.3 compatibility

### CODE REVIEW

**Comment by njzjz** [comment]
created by mistake. See #725.
